src/test/java/games/strategy/triplea/attachments/TriggerAttachmentTest.java

```diff
@@ -0,0 +1,104 @@
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.experimental.extensions.MockitoExtension;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+
+import games.strategy.engine.data.CompositeChange;
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.ProductionFrontier;
+import games.strategy.engine.data.ProductionFrontierList;
+import games.strategy.engine.data.ProductionRule;
+import games.strategy.engine.data.ProductionRuleList;
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.engine.history.IDelegateHistoryWriter;
+
+@ExtendWith(MockitoExtension.class)
+public class TriggerAttachmentTest {
+
+  public Set<TriggerAttachment> satisfiedTriggers = new HashSet<>();
+  @Mock
+  public IDelegateBridge bridge;
+  @Mock
+  public GameData gameData;
+  @Mock
+  public ProductionFrontierList productionFrontierList;
+  @Mock
+  public ProductionRuleList productionRuleList;
+  @Mock
+  public ProductionFrontier productionFrontier;
+  @Mock
+  public ProductionRule productionRule1;
+  @Mock
+  public ProductionRule productionRule2;
+  @Mock
+  public ProductionRule productionRule3;
+  @Mock
+  public IDelegateHistoryWriter historyWriter;
+
+
+  @BeforeEach
+  public void setUp() {
+    final TriggerAttachment mock = mock(TriggerAttachment.class);
+    satisfiedTriggers.add(mock);
+    when(mock.getProductionRule())
+        .thenReturn(new ArrayList<>(Arrays.asList("frontier:rule1", "frontier:-rule2", "frontier:rule3")));
+    when(mock.getName()).thenReturn("mockedTriggerAttachment");
+    when(bridge.getData()).thenReturn(gameData);
+    when(bridge.getHistoryWriter()).thenReturn(historyWriter);
+    when(gameData.getProductionFrontierList()).thenReturn(productionFrontierList);
+    when(gameData.getProductionRuleList()).thenReturn(productionRuleList);
+    when(productionFrontierList.getProductionFrontier("frontier")).thenReturn(productionFrontier);
+    when(productionFrontier.getName()).thenReturn("frontierName");
+    when(productionFrontier.getRules()).thenReturn(Arrays.asList(productionRule2));
+    when(productionRuleList.getProductionRule("rule1")).thenReturn(productionRule1);
+    when(productionRuleList.getProductionRule("rule2")).thenReturn(productionRule2);
+    when(productionRuleList.getProductionRule("rule3")).thenReturn(productionRule3);
+    when(productionRule1.getName()).thenReturn("rule1Name");
+    when(productionRule2.getName()).thenReturn("rule2Name");
+    when(productionRule3.getName()).thenReturn("rule3Name");
+
+  }
+
+  @Test
+  public void testTriggerProductionFrontierEditChange() {
```

Comment:
One technique you may find useful when creating these types of characterization tests in the future is to extract the subset of code you are refactoring to a new method and test that.  That may allow you to avoid a lot of the test code required to set up the fixture just to get to the subset of production code you're actually interested in testing.